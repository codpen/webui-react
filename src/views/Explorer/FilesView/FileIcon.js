import React from "react";
import * as PropTypes from "prop-types";

const mimeClassMap = {
    "application/pdf": "fa-file-pdf-o",
    "image/jpeg": "fa-file-image-o",
    "application/rar": "fa-file-archive-o",
    "application/x-rar-compressed": "fa-file-archive-o",
    "application/zip": "fa-file-archive-o",
    "text/plain": "fa-file-text-o",
    "text/x-vcard": "fa-address-card-o"
}

function FileIcon({IsDir, MimeType}) {
    let className = mimeClassMap[MimeType];
    if (IsDir) {
        className = "fa-folder"
    }
    if (!className) className = "fa-file";

    return <i className={className + " fa fa-lg"} data-test="fileIconComponent"/>;
}

FileIcon.propTypes = {
    IsDir: PropTypes.bool.isRequired,
    MimeType: PropTypes.string.isRequired
}

export default FileIcon;