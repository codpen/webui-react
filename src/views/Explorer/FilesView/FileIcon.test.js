import React from "react";
import {shallow} from "enzyme";
import toJson from "enzyme-to-json";
import FileIcon from "./FileIcon";
import {findByTestAttr, testStore} from "../../../../Utils";


const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);

    const component = shallow(<FileIcon {...props} store={store}/>);
    return component;
};


describe('New Mount Modal', function () {
    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};

            const props = {
                IsDir: false,
                MimeType: "text/plain"
            };
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            const component = findByTestAttr(wrapper, "fileIconComponent");
            expect(component).toHaveLength(1);
        });


        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

    });
});
