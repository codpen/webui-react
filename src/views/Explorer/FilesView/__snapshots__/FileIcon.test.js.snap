// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Mount Modal renders should match snapshot 1`] = `
<i
  className="fa-file-text-o fa fa-lg"
  data-test="fileIconComponent"
/>
`;
